### PR TITLE
Update UniquenessKey for when Element is/isn't nullable (#896)

### PR DESCRIPTION
**Pull request by zannen**
With a schema:
```gql
type Query {
  things1: [Thing] # Note the lack of "!"
}

type Subscription {
  things2: [Thing!] # Note the "!"
}
```

the `UniquenessKey` for the two lists is the same, which causes non-deterministic output.

I have:
 - [x] Added tests covering the bug (see [testing](https://github.com/99designs/gqlgen/blob/master/TESTING.md)) - **Note** I realise I have not added tests. I took a look, but as the testing doc says, it's not easy. And this Issue causes flaky output, so I'm not sure what kind of test to write. **Assistance requested**.
    * *Edit 1*: In place of a unit test, I have added a check (https://github.com/zannen/gqlgen/commit/67f8031c02aef393b134f52ce0600d75b8aa20af) for obviously different TypeReferences being overwritten in the `map`.
    * *Edit 2*: I have added an example (https://github.com/zannen/gqlgen/commit/1a4230c8655f7f8ba66ffec86bce19b5a834826e) to check for regression. I think this is sufficient to check the box.
 - [ ] Updated any relevant documentation (see [docs](https://github.com/99designs/gqlgen/tree/master/docs/content)) - **Note** I read all the docs, didn't see anywhere relevant to note this change.

### CODE REVIEW

**Comment by zannen** [comment]
I have no idea why the `plugin/resolvergen` race test is failing on CircleCI.

I ran the test over 2000 times locally and it succeeded every time:

```bash
while true ; do
  date
  go test -count=1 -race github.com/99designs/gqlgen/plugin/resolvergen
done
```

File: example/issue_896/graphql.schema [hunk 1]
```diff
@@ -0,0 +1,14 @@
+schema {
```
Comment (vektah) [inline example/issue_896/graphql.schema]: Examples are there for people to understand how gqlgen works.

This should be in `codegen/testserver/issue896.graphql` and `extend schema {`. There should be plenty of examples in that directory.

Comment (zannen) [reply example/issue_896/graphql.schema]: Thanks for the pointer, I have moved the schema file and added `extend` as you suggest.

**Comment by vektah** [comment]
Nice one!

**Pull request merged**